vsphere/folders.go

```diff
@@ -27,7 +27,7 @@ func NewFolders(client client, logger logger) Folders {
 
 // List not only lists top-level folders, it also lists child and grandchild
 // folders, and all VMs within those folders.
-func (v Folders) List(filter string, rType string) ([]common.Deletable, error) {
+func (v Folders) List(filter string, rType string, regex bool) ([]common.Deletable, error) {
```

Comment:
FWIW, I am no longer aware of any users who are using `leftovers` to delete folders on a multitenant vSphere cluster (since the Toolsmiths team no longer provides multitenant vSpheres to MAPBU). So I think you can safely implement deleting vSphere folders by regex, if that's something you want to do.

Reply:
Thanks for pointing this out. I'll need to find a good way to deal with the potential panic in `common/resource_filter.go` before we merge.